metricshub-engine/src/main/java/org/sentrysoftware/metricshub/engine/strategy/AbstractAllAtOnceStrategy.java

```diff
@@ -87,6 +88,26 @@ protected AbstractAllAtOnceStrategy(
 	 * @param hostname
 	 */
 	private void process(final Connector currentConnector, final String hostname) {
+		if (!validateConnectorDetectionCriteria(currentConnector, hostname)) {
+			log.error(
+				"Hostname {} - The connector {} no longer matches the host. Stopping the connector's collect job.",
```

Comment:
The statement `Stopping the connector's collect job` suggests it specifically pertains to the collect job, it could relate to either a simple or a discovery job. To accurately identify the job name, we should invoke the `getJobName()` method to obtain the exact job name.